packages/playwright-test/src/dispatcher.ts

```diff
@@ -262,6 +262,8 @@ export class Dispatcher {
         this._reporter.onStdErr?.('Internal error: step end without step begin: ' + params.stepId, data.test, result);
         return;
       }
+      if (params.refinedTitle)
```

Comment:
Let's add a comment that `onStepBegin` and `onStepEnd` will get different titles, and we are fine with that.

Reply:
I am not sure we should do this, let's wait for it to be reported as a concern.